### ISSUE TITLE
Remove deprecated symbols: iRODSUserGroup, UserGroup, UserGroupDoesNotExist, UserGroupManager, user_groups

### ISSUE THREAD

**Issue by paulborgermans** [irods/python-irodsclient#440]
I am not against doing the BC break, but I think there should be a clear warning in the docs there is a BC break ... or a deprecation notice and still keep (iRODS)UserGroup a reference to  (iRODS)Group (which I am using right now to deal with different PRC versions).

**Comment by trel** [comment]
Is this already broken?

Did we miss considering this during https://github.com/irods/python-irodsclient/issues/429?

**Comment by d-w-moore** [comment]
I don't think this breaks existing code unless it depends on double underscore values?

**Comment by d-w-moore** [comment]
> I don't think this breaks existing code unless it depends on double underscore values?

Or representation strings, possibly.
@paulborgermans - Is there some specific code you could point us to, that displays the breaking change?

**Comment by d-w-moore** [comment]
Ah, I think see.  The request at the heart of this issue is that we keep the  `iRODSUserGroup` alias around to avoid breaking existing code in the future.

(@paulborgermans - Please correct me if mistaken.)

**Comment by trel** [comment]
Yes, so existing code can still work until we get through the proper 'deprecation' and then 'removal' stages...

**Comment by paulborgermans** [comment]
> Ah, I think I see. The request at the heart of this issue is that we keep the `iRODSUserGroup` alias around to avoid breaking existing code in the future.
> 
> (@paulborgermans - Please correct me if mistaken.)

Yes exactly. And also UserGroup (using the model in a few queries). 

Btw I install currently python-irodsclient directly from github, hence I encountered this early before the official 1.1.7 release

its quite stable so far, frozen at the commit that was latest during the we…. Gonna take it to production by Thursday unless 1.1.7 is cut by then

**Comment by trel** [comment]
We understand the currently defined aliases will not break existing code (the new classes only provide additional functionality).

The deprecations will be declared as part of 1.1.7.

We are going to rename this issue to cover the decision about when to remove the deprecated/old classes.
